src/screens/Accounts/Accounts.js

```diff
@@ -295,39 +315,132 @@ class AccountsScreen extends React.Component<Props, State> {
+    if (smartWalletFeatureEnabled) {
+      return accounts;
+    }
+
+    return accounts.filter(({ type }) => type !== ACCOUNT_TYPES.SMART_WALLET);
+  }
+
+  wallets(activeNetwork?: BlockchainNetwork): ListItem[] {
     const {
       accounts,
-      blockchainNetworks,
-      smartWalletFeatureEnabled,
-      smartWalletState,
       navigation,
       balances,
       assets,
       rates,
+      smartWalletFeatureEnabled,
+      baseFiatCurrency,
     } = this.props;
-    const ppnNetwork = blockchainNetworks.find((network) => network.id === BLOCKCHAIN_NETWORK_TYPES.PILLAR_NETWORK);
-    const smartWalletStatus: SmartWalletStatus = getSmartWalletStatus(accounts, smartWalletState);
-    const showSmartWalletInitButton = !smartWalletStatus.hasAccount && smartWalletFeatureEnabled;
-    const visibleAccounts = smartWalletFeatureEnabled
-      ? accounts
-      : accounts.filter(({ type }) => type !== ACCOUNT_TYPES.SMART_WALLET);
-
-    const visibleAccountsWithBalance = visibleAccounts.map((acc) => {
-      const accountBalances = balances[acc.id] || {};
-      const balance = calculatePortfolioBalance(assets, rates, accountBalances);
-      return { ...acc, balance };
+
+    const visibleAccounts = this.visibleAccounts(accounts, smartWalletFeatureEnabled);
+    const hasAccount = userHasSmartWallet(accounts);
+    const showSmartWalletInitButton = !hasAccount && smartWalletFeatureEnabled;
+    const fiatCurrency = baseFiatCurrency || defaultFiatCurrency;
+
+    const isEthereumActive = !!activeNetwork && (
+      activeNetwork.id === BLOCKCHAIN_NETWORK_TYPES.ETHEREUM
+    );
+
+    const wallets = visibleAccounts.map((account: Account): ListItem => {
+      const { id, isActive, type } = account;
+      const accountBalances = balances[id];
+      const isActiveWallet = !!isActive && isEthereumActive;
+      const isSmartWallet = type === ACCOUNT_TYPES.SMART_WALLET;
+      let walletBalance = 0;
+
+      if (accountBalances) {
+        // TODO: improve balance calculation?
+        const balance = calculatePortfolioBalance(assets, rates, accountBalances);
+        walletBalance = balance && balance[fiatCurrency];
+      }
+
+      const accountItem = {
+        id: `ACCOUNT_${id}`,
+        account,
+        type: 'ACCOUNT',
+        interaction: {
+          walletBalance,
+          onMainPress: () => this.switchWallet(account),
+          onSettingsPress: () => this.accountSettings(account),
+          isActiveWallet,
+          isSmartWallet,
+        },
+      };
+
+      return accountItem;
     });
 
-    const walletsToShow = showSmartWalletInitButton
-      ? [
-        ...visibleAccountsWithBalance,
-        { type: 'SMART_WALLET_INIT', action: () => navigation.navigate(SMART_WALLET_INTRO) },
-      ]
-      : visibleAccountsWithBalance;
+    if (showSmartWalletInitButton) {
+      wallets.push({
+        type: 'NEW_SMART_WALLET',
+        interaction: {
+          isSmartWallet: true,
+          walletBalance: 0,
```

Comment:
@vyvil should we show the staked balance here?

Reply:
This kind of button wasn't designed.
As long as we are showing 0 here, it not a problem. I suppose.
(We are doing the same with PPN)
We could double check with Drew. I'll ping him :)